joplin/js/CreateContentModal/richTextPlaceholder.js

```diff
@@ -0,0 +1,51 @@
+export default function (){
+  /* - MutationObserver for Wagtail JS handling
+    - This function will handle flagged elements for custom styling and behavior.
+    - Reference: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
+      - Includes option to stop observing.
+  */
+  const wagtailFlaggedClassname = 'odd-placeholder'
+  const targetNode = document.getElementById('tab-content');
+
+  const config = { attributes: true, childList: true, subtree: true };
+
+  const callback = function(mutationsList) {
+
+    for (let mutation of mutationsList) {
+      if (mutation.type === 'childList') {
+        let placeholder = mutation.target.querySelector('.public-DraftEditorPlaceholder-inner');
+        if (placeholder) {
+          let parent = placeholder;
+          checkFlaggedAncestors(parent, placeholder);
+        }
+      }
+    }
+
+  }
+
+  function checkFlaggedAncestors(parent, placeholder){
+    // - Here, we check up through all the elements direct parent nodes (ancestors),
+    // - If a placeholder flag is present we add the class acordingly.
+    while (parent = parent.parentNode) {
+      if (
+        parent.classList &&
+        parent.classList.contains(wagtailFlaggedClassname)
+      ) {
+        parent.classList.forEach( classname => {
+          const subClasses = classname.split('_');
+          if (subClasses[0] === "odd-value") {
+            placeholder.classList.add("odd-placeolder-elm")
+            placeholder.innerText = subClasses[1].replace(/-/g," ");
```

Comment:
I'm wondering why we need the logic of parsing the name of the subclass.
Could we simplify it to (in pseudocode)
```
if (className == ''odd-option-description")
then add inner-text="Option description" <-- hardcoded here
```
I'm not sure if its necessary to get the inner-text value from the next subclass. It's a little complicated and there's a potential for bugs later on if we insert other classes there. Let me know what you think.

Reply:
Swapping it to a hard coded "Option Description" with one classname will simplifying & clarify it. I think you're right and it's just too convoluted to be add dynamic (and limited) text in wagtail to sneak through. I'll refactor for that👍